# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1383175105 14400
#      Wed Oct 30 19:18:25 2013 -0400
# Node ID ef4fc092c86b8b60660e03edf4cf2e65eee875b0
# Parent  6da521da1c700f7de39629041fdc1fc6a7bfcf04
set LC_NUMERIC local to "C" for CLI and GUI (bug #40422)

* octave.cc (octave_initialize_interpreter): Set LC_NUMERIC here.
* octave-main-thread.cc (octave_main_thread::run): Not here.

diff --git a/libgui/src/octave-main-thread.cc b/libgui/src/octave-main-thread.cc
--- a/libgui/src/octave-main-thread.cc
+++ b/libgui/src/octave-main-thread.cc
@@ -20,33 +20,29 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include <clocale>
 #include <string>
 
 #include "builtin-defun-decls.h"
 #include "octave.h"
 #include "ov-builtin.h"
 #include "ov-fcn-handle.h"
 
 #include "octave-main-thread.h"
 #include "octave-link.h"
 
 void
 octave_main_thread::run (void)
 {
-  // Matlab uses "C" locale for LC_NUMERIC class regardless of local setting
-  setlocale (LC_NUMERIC, "C");
-
   octave_initialize_interpreter (octave_cmdline_argc, octave_cmdline_argv,
                                  octave_embedded);
 
   octave_execute_interpreter ();
 }
 
 void
 octave_main_thread::execute_interpreter (void)
diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -22,16 +22,17 @@ along with Octave; see the file COPYING.
 
 // Born February 20, 1992.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cassert>
+#include <clocale>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
 
 #include <iostream>
 
 #include <fcntl.h>
 #include <getopt.h>
@@ -699,16 +700,19 @@ octave_process_command_line (int argc, c
 }
 
 // EMBEDDED is declared int instead of bool because this function is
 // declared extern "C".
 
 void
 octave_initialize_interpreter (int argc, char **argv, int embedded)
 {
+  // Matlab uses "C" locale for LC_NUMERIC class regardless of local setting
+  setlocale (LC_NUMERIC, "C");
+
   octave_embedded = embedded;
 
   octave_env::set_program_name (argv[0]);
 
   octave_program_invocation_name = octave_env::get_program_invocation_name ();
   octave_program_name = octave_env::get_program_name ();
 
   octave_thread::init ();
